### PR TITLE
Fix timeouts and simplify

### PR DESCRIPTION
**Pull request by andig**

### CODE REVIEW

File: rtuclient.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package modbus
 
 import (
 	"encoding/binary"
+	"errors"
```
Comment (premultiply) [inline rtuclient.go:9]:
```suggestion
```

File: rtuclient.go [hunk 2]
```diff
@@ -159,22 +160,21 @@ func (e *InvalidLengthError) Error() string {
...
 	var length, toRead byte
-	var n, crcCount int
-	var deadline time.Time
+	var crcCount int
 
 	for {
-		if !deadline.IsZero() && time.Now().After(deadline) { // Possible that serialport may spew data
-			return nil, fmt.Errorf("failed to read from serial port within deadline")
+		if time.Now().After(deadline) { // Possible that serialport may spew data
+			return nil, errors.New("failed to read from serial port within deadline")
```
Comment (premultiply) [inline rtuclient.go:177]:
```suggestion
			return nil, fmt.Errorf("failed to read from serial port within deadline")
```

**Review by premultiply — APPROVED** [review]
Geht nun auch ohne Faktor 2.
Warum auch immer.

**Comment by RTTTC** [comment]
I had no chance to try previous changes, but currently all my modbus intermittent errors are gone. (Used to have a timeout with EVSE DIN every couple of hours, for the last few years :D )
Heidelberg / Amperfied Energy Control wallbox also works perfectly.
Thank you guys!

**Pull request merged**